dealii@9.4.2 fails build: ShyLU/ShyLUConfig.cmake:164 (include): could not find requested file: ShyLU_NodeConfig.cmake

### Steps to reproduce the issue

`dealii@9.4.2` build fails using:
* `spack@develop` (2dc76248d3154c500fa16edfbc47801d122c7859 from `Fri Jul 7 23:27:32 2023 +0200`)
* GCC 11.1.0
* Ubuntu 20.04, X86_64

Spack environment:[spack.yaml.txt](https://github.com/spack/spack/files/11986980/spack.yaml.txt)

<details><summary>Concretization</summary>
<pre>
==> Starting concretization
==> Environment concretized in 39.40 seconds.
==> Concretized dealii
 -   ynggqa3  dealii@9.4.2%gcc@11.1.0+adol-c+arborx+arpack+assimp+cgal~cuda~doc+examples+ginkgo+gmsh+gsl+hdf5~int64~ipo+metis+mpi+muparser~nanoflann~netcdf+oce~optflags+p4est+petsc+platform-introspection~python+scalapack+simplex+slepc+sundials+symengine+threads+trilinos build_system=cmake build_type=DebugRelease cxxstd=17 generator=ninja patches=7869dfa,8a1f7b9,c9884eb arch=linux-ubuntu20.04-x86_64
 -   zxhmmxf      ^adol-c@2.7.2%gcc@11.1.0~advanced_branching+atrig_erf~boost+doc+examples~openmp~sparse+stdczero+traceless_refcounting build_system=autotools patches=715b878 arch=linux-ubuntu20.04-x86_64
 -   f6yko5w      ^arborx@1.3%gcc@11.1.0~cuda~ipo+mpi~openmp~rocm+serial~sycl+trilinos build_system=cmake build_type=Release cxxstd=17 generator=make arch=linux-ubuntu20.04-x86_64
 -   lp2ol6q          ^gmake@4.4.1%gcc@11.1.0~guile build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   gca36kf      ^arpack-ng@3.9.0%gcc@11.1.0~icb~ipo+mpi+shared build_system=cmake build_type=Release generator=make arch=linux-ubuntu20.04-x86_64
 -   cigq2yp      ^assimp@5.2.5%gcc@11.1.0~ipo+shared build_system=cmake build_type=Release generator=make arch=linux-ubuntu20.04-x86_64
 -   pkn6xtc          ^pkgconf@1.9.5%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   orihn4l      ^boost@1.79.0%gcc@11.1.0+atomic+chrono~clanglibcpp+container~context~contract~coroutine+date_time~debug+exception~fiber+filesystem+graph~graph_parallel~icu+iostreams~json+locale+log+math+mpi+multithreaded~nowide~numpy~pic+program_options~python+random+regex+serialization+shared+signals~singlethreaded+stacktrace+system~taggedlayout+test+thread+timer~type_erasure~versionedlayout+wave build_system=generic cxxstd=17 patches=a440f96,b8569d7 visibility=global arch=linux-ubuntu20.04-x86_64
 -   7yndniq          ^bzip2@1.0.8%gcc@11.1.0~debug~pic+shared build_system=generic arch=linux-ubuntu20.04-x86_64
 -   mo7lwfv          ^xz@5.4.1%gcc@11.1.0+pic build_system=autotools libs=shared,static arch=linux-ubuntu20.04-x86_64
 -   so6noth          ^zstd@1.5.5%gcc@11.1.0+programs build_system=makefile compression=none libs=shared,static arch=linux-ubuntu20.04-x86_64
 -   lh5elr2      ^cgal@5.5.2%gcc@11.1.0~core~demos+eigen~header_only~imageio~ipo+shared build_system=cmake build_type=Release generator=make arch=linux-ubuntu20.04-x86_64
 -   hwrwzj2          ^eigen@3.4.0%gcc@11.1.0~ipo build_system=cmake build_type=RelWithDebInfo generator=make arch=linux-ubuntu20.04-x86_64
 -   3kmvktv          ^gmp@6.2.1%gcc@11.1.0+cxx build_system=autotools libs=shared,static patches=69ad2e2 arch=linux-ubuntu20.04-x86_64
 -   btzlqdg              ^autoconf@2.69%gcc@11.1.0 build_system=autotools patches=35c4492,7793209,a49dd5b arch=linux-ubuntu20.04-x86_64
 -   vimk3ya              ^automake@1.16.5%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   fayzosz          ^mpfr@4.2.0%gcc@11.1.0 build_system=autotools libs=shared,static arch=linux-ubuntu20.04-x86_64
 -   l3nt23x              ^autoconf-archive@2023.02.20%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   2gbvfsg              ^texinfo@7.0.3%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   yghhaac      ^cmake@3.26.3%gcc@11.1.0~doc+ncurses+ownlibs~qt build_system=generic build_type=Release arch=linux-ubuntu20.04-x86_64
 -   6cgwuzn          ^ncurses@6.3%gcc@11.1.0~symlinks+termlib abi=none build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   q4mw6uy          ^openssl@1.1.1u%gcc@11.1.0~docs~shared build_system=generic certs=mozilla arch=linux-ubuntu20.04-x86_64
 -   w233qs7              ^ca-certificates-mozilla@2023-01-10%gcc@11.1.0 build_system=generic arch=linux-ubuntu20.04-x86_64
 -   u4ozpgd      ^ginkgo@1.6.0%gcc@11.1.0~cuda~develtools~full_optimizations~hwloc~ipo+mpi~oneapi+openmp~rocm+shared build_system=cmake build_type=Release generator=make arch=linux-ubuntu20.04-x86_64
 -   5pcyil3      ^gmsh@4.8.4%gcc@11.1.0+alglib~cairo+cgns+compression~eigen~external+fltk+gmp~hdf5~ipo+med+metis+mmg+mpi+netgen+oce~opencascade~openmp~petsc~privateapi+shared~slepc+tetgen+voropp build_system=cmake build_type=Release generator=make arch=linux-ubuntu20.04-x86_64
 -   za2fkll          ^cgns@4.4.0%gcc@11.1.0~base_scope~fortran+hdf5~int64~ipo~legacy~mem_debug+mpi+scoping+shared~static~testing~tools build_system=cmake build_type=Release generator=make arch=linux-ubuntu20.04-x86_64
 -   v3mnjza          ^fltk@1.3.7%gcc@11.1.0+gl+shared~xft build_system=generic arch=linux-ubuntu20.04-x86_64
 -   wisz4zb              ^libx11@1.8.4%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   j2cayrq                  ^inputproto@2.3.2%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   4jzlvpb                  ^kbproto@1.0.7%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   mpv4oct                  ^libxcb@1.14%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   vgnfnsn                      ^libpthread-stubs@0.4%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   rzmdgrv                      ^libxau@1.0.8%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   6jrglzo                      ^libxdmcp@1.1.4%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   v5loeht                      ^xcb-proto@1.15.2%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   ecvfe7m                  ^xextproto@7.3.0%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   rthkyjt                  ^xproto@7.0.31%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   o4feo24                  ^xtrans@1.4.0%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   a4sozci          ^freetype@2.11.1%gcc@11.1.0+pic+shared build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   awpyocw              ^libpng@1.6.39%gcc@11.1.0~ipo build_system=cmake build_type=Release generator=make libs=shared,static arch=linux-ubuntu20.04-x86_64
 -   xzytqys          ^glx@1.4%gcc@11.1.0 build_system=bundle arch=linux-ubuntu20.04-x86_64
 -   y3cnozo              ^mesa@22.1.6%gcc@11.1.0+glx+llvm+opengl~opengles+osmesa~strip build_system=meson buildtype=release default_library=shared patches=ee737d1 arch=linux-ubuntu20.04-x86_64
 -   anpwcsj                  ^bison@3.8.2%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   37tht5l                  ^flex@2.6.4%gcc@11.1.0+lex~nls build_system=autotools patches=f8b85a0 arch=linux-ubuntu20.04-x86_64
 -   4xy5bkf                      ^help2man@1.49.3%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   qnqn5uw                  ^glproto@1.4.17%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   yaaqwja                  ^libunwind@1.6.2%gcc@11.1.0~block_signals~conservative_checks~cxx_exceptions~debug~debug_frame+docs+pic+tests+weak_backtrace+xz~zlib build_system=autotools components=none libs=shared,static arch=linux-ubuntu20.04-x86_64
 -   jm5rndc                  ^libxext@1.3.3%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   4shdl6v                  ^libxt@1.1.5%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   5vws47f                      ^libice@1.0.9%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   tctga5k                      ^libsm@1.2.3%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   ny4qrnj                  ^llvm@14.0.6%gcc@11.1.0+clang~cuda~flang+gold~ipo+libomptarget~libomptarget_debug~link_llvm_dylib+lld+lldb+llvm_dylib~mlir+polly~python~split_dwarf~z3 build_system=cmake build_type=Release compiler-rt=runtime generator=ninja libcxx=runtime libunwind=runtime openmp=runtime patches=1f42874,25bc503,6379168,8248141,b216cff shlib_symbol_version=none targets=all version_suffix=none arch=linux-ubuntu20.04-x86_64
 -   osarirl                      ^binutils@2.40%gcc@11.1.0~gas+gold~gprofng+headers~interwork+ld+libiberty~lto~nls~pgo+plugins build_system=autotools compress_debug_sections=zlib libs=shared,static arch=linux-ubuntu20.04-x86_64
 -   vnxkydf                      ^libedit@3.1-20210216%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   3kmjcse                      ^perl-data-dumper@2.173%gcc@11.1.0 build_system=perl arch=linux-ubuntu20.04-x86_64
 -   p5ohhb3                  ^meson@1.1.0%gcc@11.1.0 build_system=python_pip patches=0f0b1bd arch=linux-ubuntu20.04-x86_64
 -   ooou3sj                      ^py-pip@23.1.2%gcc@11.1.0 build_system=generic arch=linux-ubuntu20.04-x86_64
 -   zmx7muq                      ^py-setuptools@67.6.0%gcc@11.1.0 build_system=generic arch=linux-ubuntu20.04-x86_64
 -   5jicjor                      ^py-wheel@0.37.1%gcc@11.1.0 build_system=generic arch=linux-ubuntu20.04-x86_64
 -   hdwxdwd                  ^py-mako@1.2.4%gcc@11.1.0 build_system=python_pip arch=linux-ubuntu20.04-x86_64
 -   heurvbh                      ^py-markupsafe@2.1.3%gcc@11.1.0 build_system=python_pip arch=linux-ubuntu20.04-x86_64
 -   a7224ek                  ^xrandr@1.5.0%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   u55qlf4                      ^libxrandr@1.5.3%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   aug2o2s                          ^renderproto@0.11.1%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   zwwa3el                      ^libxrender@0.9.10%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   6ok473p                      ^randrproto@1.5.0%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   gg5vwpx          ^med@4.0.0%gcc@11.1.0+api23~fortran~ipo+mpi~shared build_system=cmake build_type=Release generator=make patches=ba35197 arch=linux-ubuntu20.04-x86_64
 -   jvflrts          ^mesa-glu@9.0.2%gcc@11.1.0 build_system=autotools gl=glx arch=linux-ubuntu20.04-x86_64
 -   rm3x6ga          ^mmg@5.7.1%gcc@11.1.0~doc~ipo+scotch+shared~vtk build_system=cmake build_type=Release generator=make arch=linux-ubuntu20.04-x86_64
 -   eforokg              ^scotch@7.0.3%gcc@11.1.0+compression~esmumps~int64~ipo~metis+mpi~mpi_thread+shared+threads build_system=cmake build_type=Release generator=make arch=linux-ubuntu20.04-x86_64
 -   tuugpes      ^gsl@2.7.1%gcc@11.1.0~external-cblas build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   dui63eo      ^hdf5@1.10.7%gcc@11.1.0~cxx+fortran+hl~ipo~java+mpi+shared~szip~threadsafe+tools api=default build_system=cmake build_type=Release generator=make patches=2a1e311 arch=linux-ubuntu20.04-x86_64
 -   hcxjvzq      ^intel-tbb@2020.3%gcc@11.1.0+shared+tm build_system=makefile cxxstd=default patches=62ba015,ce1fb16,d62cb66 arch=linux-ubuntu20.04-x86_64
 -   dfb5iwq      ^metis@5.1.0%gcc@11.1.0~gdb~int64~ipo~real64+shared build_system=cmake build_type=Release generator=make patches=4991da9,93a7903,b1225da arch=linux-ubuntu20.04-x86_64
 -   eopcs6s      ^mpich@4.1.1%gcc@11.1.0~argobots~cuda+fortran~hwloc+hydra+libxml2+pci~rocm+romio~slurm~two_level_namespace~vci~verbs~wrapperrpath build_system=autotools datatype-engine=auto device=ch4 netmod=ofi pmi=pmi arch=linux-ubuntu20.04-x86_64
 -   brawmeh          ^findutils@4.9.0%gcc@11.1.0 build_system=autotools patches=440b954 arch=linux-ubuntu20.04-x86_64
 -   szt2cc4          ^libfabric@1.18.1%gcc@11.1.0~debug~kdreg build_system=autotools fabrics=rxm,sockets,tcp,udp arch=linux-ubuntu20.04-x86_64
 -   j5kbtgl          ^libpciaccess@0.17%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   i5p6x5b              ^util-macros@1.19.3%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   su4ha2l          ^libxml2@2.10.3%gcc@11.1.0~python build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   5f66p4a              ^libiconv@1.17%gcc@11.1.0 build_system=autotools libs=shared,static arch=linux-ubuntu20.04-x86_64
 -   or6fifa          ^yaksa@0.2%gcc@11.1.0~cuda~rocm build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   eql7hh4      ^muparser@2.3.4%gcc@11.1.0 build_system=generic arch=linux-ubuntu20.04-x86_64
 -   3d4azqo      ^netlib-scalapack@2.2.0%gcc@11.1.0~ipo~pic+shared build_system=cmake build_type=Release generator=make patches=072b006,1c9ce5f,244a9aa arch=linux-ubuntu20.04-x86_64
 -   zfif3fw      ^ninja@1.11.1%gcc@11.1.0+re2c build_system=generic arch=linux-ubuntu20.04-x86_64
 -   oh27kmh          ^python@3.10.12%gcc@11.1.0+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches=0d98e93,7d40923,f2fd060 arch=linux-ubuntu20.04-x86_64
 -   5wheyty              ^expat@2.5.0%gcc@11.1.0+libbsd build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   4lkay2l                  ^libbsd@0.11.7%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   cb26zci                      ^libmd@1.0.4%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   hq2nx75              ^gdbm@1.23%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   an65ahf              ^gettext@0.21.1%gcc@11.1.0+bzip2+curses+git~libunistring+libxml2+tar+xz build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   m2a5hf7                  ^tar@1.34%gcc@11.1.0 build_system=autotools zip=pigz arch=linux-ubuntu20.04-x86_64
 -   msq3mk6                      ^pigz@2.7%gcc@11.1.0 build_system=makefile arch=linux-ubuntu20.04-x86_64
 -   d45vpa2              ^libffi@3.4.4%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   pzz5tt3              ^libxcrypt@4.4.35%gcc@11.1.0~obsolete_api build_system=autotools patches=4885da3 arch=linux-ubuntu20.04-x86_64
 -   w3wnmrt              ^readline@8.2%gcc@11.1.0 build_system=autotools patches=bbf97f1 arch=linux-ubuntu20.04-x86_64
 -   fxjeqrt              ^sqlite@3.42.0%gcc@11.1.0+column_metadata+dynamic_extensions+fts~functions+rtree build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   lkyvaou              ^util-linux-uuid@2.38.1%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   s4ngkq3          ^re2c@2.2%gcc@11.1.0 build_system=generic arch=linux-ubuntu20.04-x86_64
 -   4ybglvi      ^oce@0.18.3%gcc@11.1.0~X11+tbb build_system=generic arch=linux-ubuntu20.04-x86_64
 -   6xmqboh      ^openblas@0.3.23%gcc@11.1.0~bignuma~consistent_fpcsr+fortran~ilp64+locking+pic+shared build_system=makefile symbol_suffix=none threads=openmp arch=linux-ubuntu20.04-x86_64
 -   e6ikymo          ^perl@5.38.0%gcc@11.1.0+cpanm+open+shared+threads build_system=generic arch=linux-ubuntu20.04-x86_64
 -   2gy24fz              ^berkeley-db@18.1.40%gcc@11.1.0+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-ubuntu20.04-x86_64
 -   4bpvbtd      ^p4est@2.8%gcc@11.1.0+mpi~openmp build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   j6swzea          ^libtool@2.4.7%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   r4bogyn      ^petsc@3.19.1%gcc@11.1.0~X~batch~cgns~complex~cuda~debug+double~exodusii~fftw+fortran~giflib+hdf5~hpddm~hwloc+hypre~int64~jpeg~knl~kokkos~libpng~libyaml~memkind+metis~mkl-pardiso~mmg~moab~mpfr+mpi~mumps~openmp~p4est~parmmg~ptscotch~random123~rocm~saws~scalapack+shared~strumpack~suite-sparse+superlu-dist~tetgen~trilinos~valgrind build_system=generic clanguage=C arch=linux-ubuntu20.04-x86_64
 -   ak3rm67          ^diffutils@3.9%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   ho2irip          ^hypre@2.28.0%gcc@11.1.0~caliper~complex~cuda~debug+fortran~gptune~int64~internal-superlu~mixedint+mpi~openmp~rocm+shared~superlu-dist~sycl~umpire~unified-memory build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   6zkypot          ^parmetis@4.0.3%gcc@11.1.0~gdb~int64~ipo+shared build_system=cmake build_type=Release generator=make patches=4f89253,50ed208,704b84f arch=linux-ubuntu20.04-x86_64
 -   7wzsy4x          ^superlu-dist@8.1.2%gcc@11.1.0~cuda~int64~ipo~openmp~rocm+shared build_system=cmake build_type=Release generator=make arch=linux-ubuntu20.04-x86_64
 -   bxftqgj      ^slepc@3.19.1%gcc@11.1.0+arpack~blopex~cuda~hpddm~rocm build_system=generic arch=linux-ubuntu20.04-x86_64
 -   lnzg7oe      ^suite-sparse@5.13.0%gcc@11.1.0~cuda~graphblas~openmp+pic build_system=generic arch=linux-ubuntu20.04-x86_64
 -   xtmngta          ^m4@1.4.19%gcc@11.1.0+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-ubuntu20.04-x86_64
 -   kcxg7oi              ^libsigsegv@2.14%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   jzcfbq6      ^sundials@6.5.1%gcc@11.1.0+ARKODE+CVODE+CVODES+IDA+IDAS+KINSOL~cuda+examples+examples-install~f2003~fcmix+generic-math~ginkgo~hypre~int64~ipo~klu~kokkos~kokkos-kernels~lapack~magma~monitoring+mpi~openmp~petsc~profiling~pthread~raja~rocm+shared+static~superlu-dist~superlu-mt~sycl~trilinos build_system=cmake build_type=Release cstd=99 cxxstd=14 generator=make logging-level=0 logging-mpi=OFF precision=double arch=linux-ubuntu20.04-x86_64
 -   twyaleu      ^symengine@0.9.0%gcc@11.1.0~boostmp~flint~ipo~llvm+mpc+mpfr~openmp~piranha+shared+thread_safe build_system=cmake build_type=Release generator=make arch=linux-ubuntu20.04-x86_64
 -   dvafxyt          ^cereal@1.3.2%gcc@11.1.0~ipo build_system=cmake build_type=Release generator=make patches=2dfa0bf arch=linux-ubuntu20.04-x86_64
 -   dkyprtg          ^mpc@1.3.1%gcc@11.1.0 build_system=autotools libs=shared,static arch=linux-ubuntu20.04-x86_64
 -   fwbxvsw      ^trilinos@13.4.1%gcc@11.1.0~adelus~adios2+amesos+amesos2+anasazi+aztec~basker+belos+boost~chaco~complex~cuda~cuda_rdc~debug~dtk+epetra+epetraext~epetraextbtf~epetraextexperimental~epetraextgraphreorderings~exodus+explicit_template_instantiation~float+fortran~gtest~hdf5~hypre+ifpack+ifpack2+intrepid+intrepid2~ipo+isorropia+kokkos~mesquite+minitensor+ml+mpi+muelu~mumps+nox~openmp~panzer+phalanx+piro~python~rocm~rocm_rdc+rol+rythmos+sacado~scorec+shards+shared+shylu+stk+stokhos+stratimikos~strumpack~suite-sparse~superlu+superlu-dist+teko+tempus+thyra+tpetra+trilinoscouplings~wrapper~x11+zoltan+zoltan2 build_system=cmake build_type=Release cxxstd=14 generator=make gotype=long_long arch=linux-ubuntu20.04-x86_64
 -   hvltmxk          ^hwloc@2.9.1%gcc@11.1.0~cairo~cuda~gl~libudev+libxml2~netloc~nvml~oneapi-level-zero~opencl+pci~rocm build_system=autotools libs=shared,static arch=linux-ubuntu20.04-x86_64
 -   vn55gu7      ^zlib@1.2.13%gcc@11.1.0+optimize+pic+shared build_system=makefile arch=linux-ubuntu20.04-x86_64
</pre>
</details>

Error:
```
$> spack -e . install
...
==> Installing dealii-9.4.2-ynggqa3bwieot6pqxgqvnhrutwufixm7
==> No binary for dealii-9.4.2-ynggqa3bwieot6pqxgqvnhrutwufixm7 found: installing from source
==> Fetching https://mirror.spack.io/_source-cache/archive/45/45a76cb400bfcff25cc2d9093d9a5c91545c8367985e6798811c5e9d2a6a6fd4.tar.gz
==> Fetching https://mirror.spack.io/_source-cache/archive/8a/8a1f7b9a155c8c496ce08b2abb1ba5d329b3b29169f36c11678aa4e3cebf97a2
==> Fetching https://mirror.spack.io/_source-cache/archive/78/7869dfab1116b6e862279bb6642c2c8fe49d87c42cfc6f031e03330f9f26a6c3
==> Fetching https://mirror.spack.io/_source-cache/archive/c9/c9884ebb0fe379c539012a225d8bcdcfe288edec8dc9d319fbfd64d8fbafba8e
==> Applied patch https://github.com/dealii/dealii/commit/06bb9dc07efb6fea9912ee0d66264af548c552c8.patch?full_index=1
==> Applied patch https://github.com/dealii/dealii/commit/40076ac1a013cd7d221f9dda913b4d0e6452c21e.patch?full_index=1
==> Applied patch https://github.com/dealii/dealii/commit/df6c5de8d6785fce701c10575982858f3aeb4cbd.patch?full_index=1
==> dealii: Executing phase: 'cmake'
==> Error: ProcessError: Command exited with status 1:
    '/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/cmake-3.26.3-yghhaaccpqxoxhtkk3iypgyq6gwhfhgi/bin/cmake' '-G' 'Ninja' '-DCMAKE_INSTALL_PREFIX:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/dealii-9.4.2-ynggqa3bwieot6pqxgqvnhrutwufixm7' '-DCMAKE_BUILD_TYPE:STRING=DebugRelease' '-DBUILD_TESTING:BOOL=OFF' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=ON' '-DCMAKE_INSTALL_RPATH:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/dealii-9.4.2-ynggqa3bwieot6pqxgqvnhrutwufixm7/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/dealii-9.4.2-ynggqa3bwieot6pqxgqvnhrutwufixm7/lib64;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/arborx-1.3-f6yko5wxowkvmg5fgfmkztcuzeojrbhr/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/arpack-ng-3.9.0-gca36kfpeu32gr66hulutrc3nuvhre65/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/assimp-5.2.5-cigq2yphtyfgg2tbqlugqzigoiqrx3ti/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/boost-1.79.0-orihn4lfw5rqiuzluavmvp33o374lpjh/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/cgal-5.5.2-lh5elr2qwsoqw7723xmy43lysmijiyag/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/ginkgo-1.6.0-u4ozpgd7sjhcl2ak2gllznhivspuciwo/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/gsl-2.7.1-tuugpestbw354efm6e3asrzfymvka6cm/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/hdf5-1.10.7-dui63eoziv72ru4keo2bbh5gyraoq77s/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/intel-tbb-2020.3-hcxjvzqe6vwgamvztdckq3l5nkp5vy73/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/metis-5.1.0-dfb5iwq2e44dpqrhzeme6robe4fy3o5b/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/mpich-4.1.1-eopcs6se47fbvf2v77wqqvw45hdoylrx/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/muparser-2.3.4-eql7hh4e5slkjaqj5hspxg4g3475owrw/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/netlib-scalapack-2.2.0-3d4azqodkwxey6c75yujj2ocsx3p64qi/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/oce-0.18.3-4ybglviwdchleqqmc3jhzdzimktxlr56/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/openblas-0.3.23-6xmqbohpydhltv7ooacyo24da6etf52k/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/p4est-2.8-4bpvbtdiluh7atoavif2r7gmghfmks6b/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/petsc-3.19.1-r4bogynzowrd23usq3yxtrbx4blx57d2/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/slepc-3.19.1-bxftqgjknezvnx45lyvkcg5drisugon2/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/suite-sparse-5.13.0-lnzg7oenvomlxvdy4sv2sskxta4kvj53/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/sundials-6.5.1-jzcfbq62al22gcyun4j6nyx7aepzwgmg/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/symengine-0.9.0-twyaleu5fz5zoxjy72bn3wm2jr6kqc25/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/trilinos-13.4.1-fwbxvswzizd34r6ebt23d664ahtypjxz/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/zlib-1.2.13-vn55gu7xzhpkptz67qfgx5ymnkyvgdke/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/adol-c-2.7.2-zxhmmxfjkgw2pk4ij7ewpghibsdeoeqn/lib64' '-DCMAKE_PREFIX_PATH:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/symengine-0.9.0-twyaleu5fz5zoxjy72bn3wm2jr6kqc25;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/mpc-1.3.1-dkyprtgqqbq7qfy22omaftdrvjqoqjev;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/cereal-1.3.2-dvafxytoxfk6jbmb2zmgfoaplhn6rvmj;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/sundials-6.5.1-jzcfbq62al22gcyun4j6nyx7aepzwgmg;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/suite-sparse-5.13.0-lnzg7oenvomlxvdy4sv2sskxta4kvj53;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/slepc-3.19.1-bxftqgjknezvnx45lyvkcg5drisugon2;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/petsc-3.19.1-r4bogynzowrd23usq3yxtrbx4blx57d2;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/hypre-2.28.0-ho2irip5ummcc6kr7qownai6ob2d3coc;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/p4est-2.8-4bpvbtdiluh7atoavif2r7gmghfmks6b;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/netlib-scalapack-2.2.0-3d4azqodkwxey6c75yujj2ocsx3p64qi;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/muparser-2.3.4-eql7hh4e5slkjaqj5hspxg4g3475owrw;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/gsl-2.7.1-tuugpestbw354efm6e3asrzfymvka6cm;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/gmsh-4.8.4-5pcyil3dhhcg3qczj5ybbd7ck2tesotg;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/oce-0.18.3-4ybglviwdchleqqmc3jhzdzimktxlr56;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/intel-tbb-2020.3-hcxjvzqe6vwgamvztdckq3l5nkp5vy73;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/ninja-1.11.1-zfif3fwyzqohcpyqxx4dk2vmsjd7qtka;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/hdf5-1.10.7-dui63eoziv72ru4keo2bbh5gyraoq77s;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/ginkgo-1.6.0-u4ozpgd7sjhcl2ak2gllznhivspuciwo;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/cgal-5.5.2-lh5elr2qwsoqw7723xmy43lysmijiyag;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/mpfr-4.2.0-fayzosznxqf5aazx3tjismzlogmrffgq;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/gmp-6.2.1-3kmvktvxixlcp552sz5u2kqd7kchsyox;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/eigen-3.4.0-hwrwzj2flhxfnbfrassd5fukuknkcr6l;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/assimp-5.2.5-cigq2yphtyfgg2tbqlugqzigoiqrx3ti;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/arpack-ng-3.9.0-gca36kfpeu32gr66hulutrc3nuvhre65;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/arborx-1.3-f6yko5wxowkvmg5fgfmkztcuzeojrbhr;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/trilinos-13.4.1-fwbxvswzizd34r6ebt23d664ahtypjxz;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/superlu-dist-8.1.2-7wzsy4xgj4znmehh6mdhbjecag4vkwyc;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/parmetis-4.0.3-6zkypotz2nrnb4i6erubku4pi7ydmhqi;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/openblas-0.3.23-6xmqbohpydhltv7ooacyo24da6etf52k;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/metis-5.1.0-dfb5iwq2e44dpqrhzeme6robe4fy3o5b;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/hwloc-2.9.1-hvltmxkxcv4nh6rdwvv566luuqt2a2tm;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/boost-1.79.0-orihn4lfw5rqiuzluavmvp33o374lpjh;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/mpich-4.1.1-eopcs6se47fbvf2v77wqqvw45hdoylrx;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/yaksa-0.2-or6fifarhx5n2biqi6hheycxa63csrgv;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/zstd-1.5.5-so6nothpdqyv4zfvxyuila56ka35vgta;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libxml2-2.10.3-su4ha2lyuhaaqogchr7taxrl5acockez;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/xz-5.4.1-mo7lwfver6ug5jejkwt6gkjog76k7lxc;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libpciaccess-0.17-j5kbtglykqfohsexjzn5xgtkw27tf3kj;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libfabric-1.18.1-szt2cc4tspsr5wb56qtobdlf6iabycoq;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/cmake-3.26.3-yghhaaccpqxoxhtkk3iypgyq6gwhfhgi;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/zlib-1.2.13-vn55gu7xzhpkptz67qfgx5ymnkyvgdke;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/bzip2-1.0.8-7yndniqspntmfkwblnzygqiv5ptmdlrx;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libiconv-1.17-5f66p4aoivmlxa7htmpsffk5ov62vgci;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/ncurses-6.3-6cgwuznb5y3wijnhs2tpdcnwviui5cur;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/adol-c-2.7.2-zxhmmxfjkgw2pk4ij7ewpghibsdeoeqn' '-DDEAL_II_EXAMPLES_RELDIR:STRING=share/deal.II/examples' '-DDEAL_II_DOCREADME_RELDIR:STRING=share/deal.II/' '-DDEAL_II_DOCHTML_RELDIR:STRING=share/deal.II/doc' '-DBOOST_DIR:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/boost-1.79.0-orihn4lfw5rqiuzluavmvp33o374lpjh' '-DLAPACK_FOUND:BOOL=ON' '-DLAPACK_INCLUDE_DIRS:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/openblas-0.3.23-6xmqbohpydhltv7ooacyo24da6etf52k/include' '-DLAPACK_LIBRARIES:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/openblas-0.3.23-6xmqbohpydhltv7ooacyo24da6etf52k/lib/libopenblas.so' '-DUMFPACK_DIR:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/suite-sparse-5.13.0-lnzg7oenvomlxvdy4sv2sskxta4kvj53' '-DZLIB_DIR:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/zlib-1.2.13-vn55gu7xzhpkptz67qfgx5ymnkyvgdke' '-DDEAL_II_ALLOW_BUNDLED:BOOL=OFF' '-DDEAL_II_COMPONENT_DOCUMENTATION:BOOL=OFF' '-DDEAL_II_COMPONENT_EXAMPLES:BOOL=ON' '-DDEAL_II_WITH_64BIT_INDICES:BOOL=OFF' '-DDEAL_II_WITH_CUDA:BOOL=OFF' '-DDEAL_II_WITH_MPI:BOOL=ON' '-DCMAKE_C_COMPILER:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/mpich-4.1.1-eopcs6se47fbvf2v77wqqvw45hdoylrx/bin/mpicc' '-DCMAKE_CXX_COMPILER:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/mpich-4.1.1-eopcs6se47fbvf2v77wqqvw45hdoylrx/bin/mpic++' '-DCMAKE_Fortran_COMPILER:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/mpich-4.1.1-eopcs6se47fbvf2v77wqqvw45hdoylrx/bin/mpif90' '-DMPI_C_COMPILER:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/mpich-4.1.1-eopcs6se47fbvf2v77wqqvw45hdoylrx/bin/mpicc' '-DMPI_CXX_COMPILER:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/mpich-4.1.1-eopcs6se47fbvf2v77wqqvw45hdoylrx/bin/mpic++' '-DMPI_Fortran_COMPILER:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/mpich-4.1.1-eopcs6se47fbvf2v77wqqvw45hdoylrx/bin/mpif90' '-DDEAL_II_COMPONENT_PYTHON_BINDINGS:BOOL=OFF' '-DDEAL_II_WITH_TBB:BOOL=ON' '-DTBB_DIR:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/intel-tbb-2020.3-hcxjvzqe6vwgamvztdckq3l5nkp5vy73' '-DDEAL_II_WITH_GSL:BOOL=ON' '-DGSL_DIR:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/gsl-2.7.1-tuugpestbw354efm6e3asrzfymvka6cm' '-DDEAL_II_WITH_HDF5:BOOL=ON' '-DHDF5_DIR:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/hdf5-1.10.7-dui63eoziv72ru4keo2bbh5gyraoq77s' '-DDEAL_II_WITH_P4EST:BOOL=ON' '-DP4EST_DIR:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/p4est-2.8-4bpvbtdiluh7atoavif2r7gmghfmks6b' '-DDEAL_II_WITH_PETSC:BOOL=ON' '-DPETSC_DIR:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/petsc-3.19.1-r4bogynzowrd23usq3yxtrbx4blx57d2' '-DDEAL_II_WITH_SLEPC:BOOL=ON' '-DSLEPC_DIR:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/slepc-3.19.1-bxftqgjknezvnx45lyvkcg5drisugon2' '-DDEAL_II_WITH_TRILINOS:BOOL=ON' '-DTRILINOS_DIR:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/trilinos-13.4.1-fwbxvswzizd34r6ebt23d664ahtypjxz' '-DDEAL_II_WITH_METIS:BOOL=ON' '-DMETIS_DIR:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/metis-5.1.0-dfb5iwq2e44dpqrhzeme6robe4fy3o5b' '-DDEAL_II_WITH_SUNDIALS:BOOL=ON' '-DSUNDIALS_DIR:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/sundials-6.5.1-jzcfbq62al22gcyun4j6nyx7aepzwgmg' '-DDEAL_II_WITH_NANOFLANN:BOOL=OFF' '-DDEAL_II_WITH_ASSIMP:BOOL=ON' '-DASSIMP_DIR:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/assimp-5.2.5-cigq2yphtyfgg2tbqlugqzigoiqrx3ti' '-DDEAL_II_WITH_GMSH:BOOL=ON' '-DGMSH_DIR:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/gmsh-4.8.4-5pcyil3dhhcg3qczj5ybbd7ck2tesotg' '-DDEAL_II_WITH_MUPARSER:BOOL=ON' '-DMUPARSER_DIR:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/muparser-2.3.4-eql7hh4e5slkjaqj5hspxg4g3475owrw' '-DDEAL_II_WITH_SYMENGINE:BOOL=ON' '-DSYMENGINE_DIR:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/symengine-0.9.0-twyaleu5fz5zoxjy72bn3wm2jr6kqc25' '-DDEAL_II_WITH_GINKGO:BOOL=ON' '-DGINKGO_DIR:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/ginkgo-1.6.0-u4ozpgd7sjhcl2ak2gllznhivspuciwo' '-DDEAL_II_WITH_ARBORX:BOOL=ON' '-DARBORX_DIR:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/arborx-1.3-f6yko5wxowkvmg5fgfmkztcuzeojrbhr' '-DDEAL_II_WITH_CGAL:BOOL=ON' '-DCGAL_DIR:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/cgal-5.5.2-lh5elr2qwsoqw7723xmy43lysmijiyag' '-DDEAL_II_WITH_ADOLC:BOOL=ON' '-DADOLC_DIR:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/adol-c-2.7.2-zxhmmxfjkgw2pk4ij7ewpghibsdeoeqn' '-DDEAL_II_WITH_ARPACK:BOOL=ON' '-DARPACK_DIR:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/arpack-ng-3.9.0-gca36kfpeu32gr66hulutrc3nuvhre65' '-DDEAL_II_ARPACK_WITH_PARPACK:BOOL=ON' '-DDEAL_II_WITH_NETCDF:BOOL=OFF' '-DDEAL_II_WITH_SCALAPACK:BOOL=ON' '-DSCALAPACK_FOUND:BOOL=ON' '-DSCALAPACK_INCLUDE_DIRS:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/netlib-scalapack-2.2.0-3d4azqodkwxey6c75yujj2ocsx3p64qi/include' '-DSCALAPACK_LIBRARIES:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/netlib-scalapack-2.2.0-3d4azqodkwxey6c75yujj2ocsx3p64qi/lib/libscalapack.so' '-DSCALAPACK_LIBRARY:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/netlib-scalapack-2.2.0-3d4azqodkwxey6c75yujj2ocsx3p64qi/lib/libscalapack.so' '-DDEAL_II_WITH_OPENCASCADE:BOOL=ON' '-DOPENCASCADE_DIR:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/oce-0.18.3-4ybglviwdchleqqmc3jhzdzimktxlr56' '-DCMAKE_CXX_FLAGS:STRING=-std=c++17' '-DDEAL_II_CXX_FLAGS:STRING=-march=x86-64 -mtune=generic' '-DDEAL_II_ALLOW_PLATFORM_INTROSPECTION:BOOL=ON' '/tmp/root/spack-stage/spack-stage-dealii-9.4.2-ynggqa3bwieot6pqxgqvnhrutwufixm7/spack-src'

1 error found in build log:
     372      cmake/macros/macro_configure_feature.cmake:237 (EVALUATE_EXPRESSION)
     373      cmake/configure/configure_20_trilinos.cmake:428 (CONFIGURE_FEATURE)
     374      cmake/macros/macro_verbose_include.cmake:19 (INCLUDE)
     375      CMakeLists.txt:130 (VERBOSE_INCLUDE)
     376
     377
  >> 378    CMake Error at /spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/trilinos-13.4.1-fwbxvswzizd34r6ebt23d664ahtypjxz/lib/cmake/ShyLU/ShyLUConfig.cmake:164 (include):
     379      include could not find requested file:
     380
     381        /spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/trilinos-13.4.1-fwbxvswzizd34r6ebt23d664ahtypjxz/lib/cmake/ShyLU/../ShyLU_Node/ShyLU_NodeConfig.cmake
     382    Call Stack (most recent call first):
     383      /spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/trilinos-13.4.1-fwbxvswzizd34r6ebt23d664ahtypjxz/lib/cmake/Trilinos/TrilinosConfig.cmake:130 (include)
     384      cmake/modules/FindTRILINOS.cmake:38 (FIND_PACKAGE)
```

### Error message

<details><summary>Error message</summary>
<pre>
... see above
</pre></details>


### Information on your system

* **Spack:** 0.21.0.dev0 (2dc76248d3154c500fa16edfbc47801d122c7859)
* **Python:** 3.8.10
* **Platform:** linux-ubuntu20.04-cascadelake
* **Concretizer:** clingo

### Additional information

[spack-build-env.txt](https://github.com/spack/spack/files/11987219/spack-build-env.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/11987220/spack-build-out.txt)

@jppelteret @luca-heltai @wspear

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/share/spack/gitlab/cloud_pipelines/stacks/e4s/spack.yaml
+++ b/share/spack/gitlab/cloud_pipelines/stacks/e4s/spack.yaml
@@ -153,7 +153,7 @@ spack:
   - variorum
   - veloc
   - wannier90
-  - xyce +mpi +shared +pymi +pymi_static_tpls ^trilinos~shylu
+  - xyce +mpi +shared +pymi +pymi_static_tpls ^trilinos +shylu
 
   # CUDA
   - amrex +cuda

--- a/var/spack/repos/builtin/packages/trilinos/package.py
+++ b/var/spack/repos/builtin/packages/trilinos/package.py
@@ -447,6 +447,8 @@ class Trilinos(CMakePackage, CudaPackage, ROCmPackage):
 
     # ###################### Patches ##########################
 
+    patch("shylu-node-optional.patch", when="@13:14.4.0 +shylu")
+
     patch("umfpack_from_suitesparse.patch", when="@11.14.1:12.8.1")
     for _compiler in ["xl", "xl_r", "clang"]:
         patch("xlf_seacas.patch", when="@12.10.1:12.12.1 %" + _compiler)

--- a/var/spack/repos/builtin/packages/trilinos/shylu-node-optional.patch
+++ b/var/spack/repos/builtin/packages/trilinos/shylu-node-optional.patch
@@ -1,0 +1,10 @@
+diff -ruN spack-src/packages/shylu/cmake/Dependencies.cmake spack-src-patched/packages/shylu/cmake/Dependencies.cmake
+--- spack-src/packages/shylu/cmake/Dependencies.cmake	2023-08-01 15:08:54.000000000 -0700
++++ spack-src-patched/packages/shylu/cmake/Dependencies.cmake	2023-09-22 12:53:28.755219076 -0700
+@@ -1,4 +1,5 @@
+-SET(LIB_REQUIRED_DEP_PACKAGES ShyLU_DD ShyLU_Node)
++SET(LIB_REQUIRED_DEP_PACKAGES ShyLU_DD)
++SET(LIB_OPTIONAL_DEP_PACKAGES ShyLU_Node)
+ SET(LIB_OPTIONAL_DEP_PACKAGES)
+ SET(TEST_REQUIRED_DEP_PACKAGES)
+ SET(TEST_OPTIONAL_DEP_PACKAGES)

